htdocs/user/messaging.php

```diff
@@ -86,18 +86,13 @@
 // Initialize technical object to manage hooks of page. Note that conf->hooks_modules contains array of hook context
 $hookmanager->initHooks(array('agendathirdparty', 'globalcard'));
 
-// Security check
-if ($user->id) {
```

Comment:
I don't think we must remove this but move it before the 
```
// Initialize technical objects
$object = new User($db);
if ($id > 0 || !empty($ref)) {
```

So an external user will always be forced to himself